### PR TITLE
Allow AutoSearchInput to be called without a location property

### PR DESCRIPTION
**Pull request by bobsilverberg**
Supports #3991

I need to be able to prevent the AutoSearchInput component from automatically creating filters based on the location query parameters for my changes to the CollectionManager to support sorting add-ons in a collection, and this seems like a simple way of doing that.

### CODE REVIEW

File: tests/unit/amo/components/TestAutoSearchInput.js [hunk 1]
```diff
@@ -297,6 +297,28 @@ describe(__filename, () => {
...
+        store,
+      });
+
+      const query = 'ad blocker';
+      fetchSuggestions({ root, query });
+
+      sinon.assert.calledWith(
+        dispatch,
+        autocompleteStart({
+          errorHandlerId: root.instance().props.errorHandler.id,
```
Comment (willdurand) [inline tests/unit/amo/components/TestAutoSearchInput.js]: It would be better to use `createStubErrorHandler()` and pass it to `render()`.

**Comment by bobsilverberg** [comment]
> How do you want to use this component without a location because it uses the withRouter() HOC, which (always?) injects the location.

I guess that's what I guess for separating this change from the code that will use it. I didn't test this new version with my code that needs it. 

I didn't realize that it automatically gets a `location` injected because we do pass a `location` in when it is called: 

- https://github.com/mozilla/addons-frontend/blob/d9b067d4a53c25821d6defc2eb75583ac4892f1f/src/amo/components/SearchForm/index.js#L58
- https://github.com/mozilla/addons-frontend/blob/b8a86fce7ba51ed7681ad05354f0dfa9f6af1932/src/amo/components/CollectionManager/index.js#L412

which seems odd.

I'll test this patch with my other code to see if it works as expected.

File: tests/unit/amo/components/TestAutoSearchInput.js (hunk 1 above)
Comment (bobsilverberg) [reply tests/unit/amo/components/TestAutoSearchInput.js]: Yes, I really just copied an existing test and only made the changes necessary to it.

**Comment by willdurand** [comment]
> I didn't realize that it automatically gets a location injected because we do pass a location in when it is called:

ah, then the `withRouter` HOC should probably be removed.

**Comment by bobsilverberg** [comment]
Removing `withRouter` fixes it for me, and if we're always passing in a `location` (except when we don't want it to use the `location`) then we don't need `withRouter` anymore. I've updated the PR to remove it.

**Comment by bobsilverberg** [comment]
Sorry, @willdurand, I'm getting a bit flustered trying to juggle all these PRs I'm working on and teasing apart, so I did a `git commit --amend` where I shouldn't have so the change to remove `withRouter` got folded into the original commit. I did fix the test in a separate commit though. This should be ready for another look.

**Pull request merged**